jellyfish.livemd

```diff
@@ -81,41 +77,105 @@ Add Jellyfish SDK to Elixir deps in `mix.exs`:
+2. Create a new [`Phoenix.Channel`](https://hexdocs.pm/phoenix/channels.html), which will be responsible for the communication browser and backend.
+3. On join, send a message to `self` and let the user join.
+4. Handle the message sent above and create a new [`Jellyfish.Client`](https://hexdocs.pm/jellyfish_server_sdk/Jellyfish.Client.html) and use it to create a new [`Jellyfish.Room`](https://hexdocs.pm/jellyfish_server_sdk/Jellyfish.Room.html). Then add a new `webtrc` peer to the jellyfish room. Remember to assign `client`, `room_id`, and `peer_id` to the socket.
+5. From the above callback push the `PeerToken` over the socket.
+6. We should be handling the termination of the room - we will handle it better later, for now, we may simply call `Room.delete` on the channel exit.
 7. Go to the JS part of the channel and handle the token message.
 
+If you correctly implement all points above you should receive a token on fronted and could log it.
+
 ## Frontend - let's send some media to Jellyfish
 
-The next step is to use the obtained token to start sending media to Jellyfish. We're going to use JS WebRTC APIs and [Jellyfish TS client](https://github.com/jellyfish-dev/jellyfish-react-client/blob/update-readme/src/jellyfish/readme.md).
+The current state is that on jellyfish is created room and peer and also the client (browser) received a `PeerToken`.
+The next step is to use this token to successfully connect to Jellyfish and start sending media to it. 
+Then we will have to add handling for messages received from Jellyfish. 
+We're going to use JS WebRTC APIs and [Jellyfish TS client](https://jellyfish-dev.github.io/ts-client-sdk/). 
+This client SDK notifies the browser about situations like a new peer joining or receiving a new track.
+
+First things first we have to initialize our jellyfish TS client, it will interact with your js app, through callbacks, that's why implementing them will be our next task.
+There are [many callbacks](https://jellyfish-dev.github.io/ts-client-sdk/interfaces/MessageEvents.html) that you can implement, but now we can focus on a few of them.
+Remember to define callbacks before connecting to the jellyfish instance to avoid race conditions.
+So the first callback will handle a situation when a peer successfully joins the room, this is a moment when we can start sending media through Webrtc.
+Similarly to the previous exercise, we have to acquire our own camera and display it locally and then add it to our jellyfish client.
+Another thing to do is prepare yourself for receiving videos from other peers, for that we have to create a video element for each peer that is in a room or will join the room, also we have to attach ready tracks to proper video elements.
```

Comment:
```suggestion
The next step is to prepare yourself for receiving videos from other peers. For that we have to create a video element for each peer that is in a room or will join the room, also we have to attach ready tracks to proper video elements.
```

Reply:
We should probably add audio HTML as well, so that lack of video won't block playing the audio

Reply:
Fair point, I followed too much our [dashboard implementation](https://github.com/jellyfish-dev/ts-client-sdk/blob/main/examples/simple-app/src/main.ts). Do we want that in the basic version of the workshop, will be required to implementing the waiting room? Because if a peer can't decide if the camera is switched on/off before joining the room, I think that we can assume that all peers will join with the camera and audio on.